app/src/main/java/de/codefor/karlsruhe/opensense/widget/base/BaseWidgetConfigurationActivity.kt

```diff
@@ -31,31 +39,57 @@ abstract class BaseWidgetConfigurationActivity : AppCompatActivity() {
     private var boxId = ""
 
     private lateinit var coordinatorLayout: CoordinatorLayout
-    private lateinit var boxIdEditText: EditText
     private lateinit var boxInfoLayout: LinearLayout
     private lateinit var boxName: TextView
     private lateinit var boxDescription: TextView
     private lateinit var boxSensorsRecyclerView: RecyclerView
 
-    private var addWidgetOnClickListener: View.OnClickListener = View.OnClickListener {
-        WidgetHelper.getSenseBox(boxIdEditText.text.toString())
-                .subscribe(this::showBoxInformation) {
-                    Snackbar.make(coordinatorLayout, R.string.widget_configuration_snackbar_error_loading, Snackbar.LENGTH_SHORT)
-                            .show()
-                }
-    }
+    private lateinit var mapView: MapView
+
+    private lateinit var mapboxMap: MapboxMap
+
 
     public override fun onCreate(icicle: Bundle?) {
         super.onCreate(icicle)
 
         setResult(Activity.RESULT_CANCELED)
 
+        // Mapbox Access token
+        Mapbox.getInstance(getApplicationContext(), getResources().getString(R.string.mapbox_token));
+
         setContentView(R.layout.activity_base_widget_configuration)
 
-        findViewById<View>(R.id.default_widget_configure_select).setOnClickListener(addWidgetOnClickListener)
+
+        mapView = findViewById<View>(R.id.mapView) as MapView
+        mapView.onCreate(icicle)
+
+        mapView.getMapAsync { mapboxMap ->
+            run {
+                this.mapboxMap = mapboxMap
+
+                // get all boxes from api and display them on the map
+                WidgetHelper.getAllBoxes()
+                        .subscribe(this::displayBoxesOnMap)
```

Comment:
I was not able to figure out how to pass parameters to the subscription method, therefore I created a variable containing the `mapboxMap`

Reply:
You could do it this way: `WidgetHelper.getAllBoxes().subscribe({boxes -> displayBoxesOnMap(mapboxMap, boxes)})` and add the MapboxMap argument to the method. But don't trouble yourself. If you like, I'll change it later on.